zproject/backends.py

```diff
@@ -422,6 +429,39 @@ def authenticate(self, username, realm_subdomain=None, return_data=None):
 class GitHubAuthBackend(SocialAuthMixin, GithubOAuth2):
     auth_backend_name = u"GitHub"
 
+    def authenticate(self,
+                     realm_subdomain='',  # type: Optional[Text],
+                     storage=None,  # type: Optional[social_django.models.DjangoStorage]
+                     strategy=None,  # type: Optional[social_django.strategy.DjangoStrategy]
+                     user=None,  # type: Optional[Dict[str, Any]],
+                     return_data=None,  # type: Optional[Dict[str, Any]]
+                     response=None,  # type: Optional[Dict[str, Any]]
+                     backend=None  # type: Optional[GithubOAuth2]
+                     ):
+        # type: (...) -> Optional[UserProfile]
+        """
+        Django decides which `authenticate` to call by inspecting the
+        arguments. So it's better to create `authenticate` function
+        with well defined arguments.
+        """
+        if user is None:
+            user = {}
+
+        if return_data is None:
+            return_data = {}
+
+        if response is None:
+            response = {}
+
+        return self.common_authenticate(self,
+                                        realm_subdomain=realm_subdomain,
+                                        storage=storage,
+                                        strategy=strategy,
+                                        user=user,
+                                        return_data=return_data,
+                                        response=response,
+                                        backend=backend)
+
```

Comment:
It can. So I was confused about where to put this. I decided on this approach because we can add a new backend whose `authenticate` function accepts different arguments. This way every new backend can define its own `authenticate` method and we won't have to update the code in `common_authenticate`.